flink-streaming-java/src/test/java/org/apache/flink/streaming/runtime/tasks/StreamTaskTest.java

```diff
@@ -945,12 +982,15 @@ public void testAsyncCheckpointingConcurrentCloseBeforeAcknowledge() throws Exce
         // make sure that all state handles have been discarded
         discardFuture.get();
 
-        try {
-            mockEnvironment.getAcknowledgeCheckpointFuture().get(10L, TimeUnit.MILLISECONDS);
-            fail("The checkpoint should not get acknowledged.");
-        } catch (TimeoutException expected) {
-            // future should not be completed
-        }
+        assertThatThrownBy(
+                        () -> {
+                            // future should not be completed
+                            mockEnvironment
+                                    .getAcknowledgeCheckpointFuture()
+                                    .get(10L, TimeUnit.MILLISECONDS);
+                            fail("The checkpoint should not get acknowledged.");
```

Comment:
```suggestion
```

Can this line be removed? I remember  if it doesn't throw expected exception, `assertThatThrownBy` will fail, right?

Reply:
Yes. I'll check the same cases in the test.
thx~